viewing_party/party.py

```diff
@@ -0,0 +1,243 @@
+        return None
+
+    for movie_dict in user_data["watched"]:
+        genre_list.append(movie_dict["genre"])
+
+    fantasy_count = genre_list.count("Fantasy")
+    action_count = genre_list.count("Action")
+    intrigue_count = genre_list.count("Intrigue")
+
+    if (fantasy_count > action_count) and (fantasy_count > intrigue_count):
+        return "Fantasy"
+    elif (action_count > fantasy_count) and (action_count > intrigue_count):
+        return "Action"
+    else: 
+        return "Intrigue"
+
+
+
+
+
+
+
+
+################ WAVE 3 ##############
+
+#wave 3, part 1
+def get_unique_watched(user_data):
+    """Takes dict, returns list of dicts that only user has seen."""
+    
+    only_user_watched = []
+    user_watched_set = set()
+    friends_watched_set = set()
+    only_user_watched_set = set()
+
+    for movie_user_watched in user_data["watched"]:
+        user_watched_set.add(movie_user_watched["title"])
+        for friends_watched_lists in user_data["friends"]:
+            for movies_friends_watched_list in friends_watched_lists.values():
+                for indiv_dict in movies_friends_watched_list:
+                    friends_watched_set.add(indiv_dict["title"])
+
+    only_user_watched_set = user_watched_set.difference(friends_watched_set)
+
+    for item in only_user_watched_set:
+        only_user_watched.append({"title" : item})
+
+    return only_user_watched
+                    
+    
+
+#wave 3, part 2
+def get_friends_unique_watched(user_data):
+    """Takes dict, returns list of dicts that only friends have seen."""
+
+    user_watched_set = set()
+    friends_watched_set = set()
+    only_friends_watched_set = set()
+    only_friends_watched = []
+
+    for movie_user_watched in user_data["watched"]:
+        user_watched_set.add(movie_user_watched["title"])
+        
+    for friends_watched_lists in user_data["friends"]:
+        for movies_friends_watched_list in friends_watched_lists.values():
+            for indiv_dict in movies_friends_watched_list:
+                friends_watched_set.add(indiv_dict["title"])
+
+    only_friends_watched_set = friends_watched_set.difference(user_watched_set)
+
+    for item in only_friends_watched_set:
+        only_friends_watched.append({"title" : item})
+
+    return only_friends_watched
+
+
+
+
+################ WAVE 4 ##############
+
+def get_available_recs(user_data):
+    """Takes dict, returns list of movie recommendations."""
+
+    friends_watched_list = get_friends_unique_watched(user_data)
+    friends_watched_list_titles = []
+    rec_list = []
+    
+    for item in friends_watched_list:
+        friends_watched_list_titles.append(item["title"])
+
+    for watched_lists in user_data["friends"]:
+        for watched_dict in watched_lists["watched"]:
+            if (watched_dict["host"] in user_data["subscriptions"]) \
+                and (watched_dict not in rec_list) \
+                and (watched_dict["title"] in friends_watched_list_titles):
```

Comment:
With some changes to the how we use parentheses here, we can avoid the explicit line wrapping, like:

```suggestion
            if (watched_dict["host"] in user_data["subscriptions"]
                and watched_dict not in rec_list
                and watched_dict["title"] in friends_watched_list_titles):
```